### PR TITLE
PLATFORM-2466 Set tracer flag if sjc-backend is unhealthy

### PR DESCRIPTION
**Pull request by pgrabowski-wikia**
https://wikia-inc.atlassian.net/browse/PLATFORM-2466

- [x] If there is no header sent, lets just not include it in logs (It's sent only from SJC datacenter)	
- [x] Log only when header says backend is unhealthy

//cc
@Grunny @themech

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [WikiaTracer.php#L7](https://github.com/Wikia/app/blob/3cc81c6078610aae0218344134be3dc13bcb48f2/lib/Wikia/src/Tracer/WikiaTracer.php#L7): Class "WikiaTracer" has 29 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
2. MAJOR [WikiaTracer.php#L370](https://github.com/Wikia/app/blob/3cc81c6078610aae0218344134be3dc13bcb48f2/lib/Wikia/src/Tracer/WikiaTracer.php#L370): Remove the unused function parameter "$cmd". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
3. MAJOR [WikiaTracer.php#L391](https://github.com/Wikia/app/blob/3cc81c6078610aae0218344134be3dc13bcb48f2/lib/Wikia/src/Tracer/WikiaTracer.php#L391): Remove the unused function parameter "$method". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

**Review by Grunny — APPROVED** [review]
LGTM.

File: lib/Wikia/src/Tracer/WikiaTracer.php [hunk 1]
```diff
@@ -12,6 +12,8 @@ class WikiaTracer {
 	const PARENT_SPAN_ID_HEADER_NAME = 'X-Parent-Span-Id';
 	const CLIENT_IP_HEADER_NAME = 'X-Client-Ip';
 	const SHIELDS_HEADER_NAME = 'X-SJC-shields-healthy';
+	const SHIELDS_HEADER_STATUS_HEALTHY = '1';
```
Comment (themech) [inline lib/Wikia/src/Tracer/WikiaTracer.php:15]: I don't think it is used at the moment.

File: lib/Wikia/src/Tracer/WikiaTracer.php [hunk 2]
```diff
@@ -310,7 +314,7 @@ public function getPublicHeaders() {
 			self::LEGACY_TRACE_ID_HEADER_NAME => $this->traceId,
 			// pass the current span ID to the subrequest, it will be logged as parent_span_id there
 			self::PARENT_SPAN_ID_HEADER_NAME => $this->spanId,
-			self::SHIELDS_HEADER_NAME => $this->sjcShields,
+			self::SHIELDS_HEADER_NAME => $this->sjcShieldsHealthStatus,
```
Comment (themech) [inline lib/Wikia/src/Tracer/WikiaTracer.php:317]: sjcShieldsHealthStatus containt 1 if the shields are unhealthy. Isn't it misleading that we flipped the value here, so the "X-SJC-shields-healthy" is set to 1 if the status is unhealthy?

Comment (pgrabowski-wikia) [reply lib/Wikia/src/Tracer/WikiaTracer.php:317]: continue: https://github.com/Wikia/app/pull/11871

**Pull request merged**